xarray/core/coordinates.py

```diff
@@ -314,6 +315,71 @@ def __unicode__(self):
         return formatting.indexes_repr(self)
 
 
+def normalize_indexes(indexes, coords, sizes):
```

Comment:
I'm trying to follow this PR up, but not yet sure what this utility function does.

Can you suggest an example usage of this?
What I currently think of is 
+ Without MultiIndex
```python
    coords = OrderedDict()
    coords['x'] = Variable(('x', ), [0, 1, 2])
    coords['y'] = Variable(('y', ), ['a', 'b'])
    sizes = {'x': 3, 'y': 2}
    actual = normalize_indexes({}, coords, sizes)
    expected = {'x': coords['x'].to_index(), 'y': coords['y'].to_index()}
```
+ With MultiIndex
```python
    coords = OrderedDict()
    coords['x'] = Variable(('z', ), [0, 1, 2])
    coords['y'] = Variable(('z', ), ['a', 'b'])
    sizes = {'x': 6}
    actual = normalize_indexes({}, coords, sizes)
    expected = {'x': pd.Index([0, 0, 1, 1, 2, 2]),
                'y': pd.Index(['a', 'b', 'a', 'b', 'a', 'b'])}
```
Or do we need a `MultiIndex` for `'z'` ?

Reply:
Your "without MultiIndex" example look correct, but I think your "with MultiIndex" case should raise an error, because the dimension "z" has inconsistent sizes.

I was thinking something more like:
```
    indexes = {
        'x': pandas.Index([0, 0, 1, 1, 2, 2]),
        'y': pandas.Index(['a', 'b', 'a', 'b', 'a', 'b']),
    }
    coords = {
        'x': Variable(('z',), [0, 0, 1, 1, 2, 2]),
        'y': Variable(('z',), ['a', 'b', 'a', 'b', 'a', 'b']),
    }
    sizes = {'z': 6}
    actual = normalize_indexes(indexes, coords, sizes)
    combined_index = pandas.MultiIndex.from_arrays([indexes['x'], indexes['y']], names=['x', 'y'])
    expected = {'x': combined_index, 'y': combined_index, 'z': combined_index}
```

The resulting object would supporting indexing with any of `.sel(x=...)`, `.sel(y=...)` or `.sel(z=...)`, all based on `combined_index`.